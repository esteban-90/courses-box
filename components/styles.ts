import { css, SerializedStyles } from '@emotion/react'
import { AppTheme } from '@/styles/themes'

export type Color = 'primary' | 'secondary' | 'danger' | 'warning'

export const boxShadow = (shadowColor1: string, shadowColor2: string, inset = false): SerializedStyles => {
  const insetStr = inset ? 'inset' : false

  return css`
    box-shadow: 0.5vmin 0.5vmin 1vmin ${shadowColor1} ${insetStr}, -0.5vmin -0.5vmin 1vmin ${shadowColor2} ${insetStr};
  `
}

export const transition = (): SerializedStyles => {
  return css`
    transition: all 0.4s ease;
  `
}

export const getColors = (theme: AppTheme, color?: Color): SerializedStyles => {
  switch (color) {
    case 'secondary':
      return css`
        color: ${theme.font.regular};
      `

    case 'primary':
    case 'danger':
      return css`
        background: ${theme.components[color]};
        color: ${theme.font.button};
      `

    case 'warning':
      return css`
        background: ${theme.components[color]};
        color: ${theme.font.warning};
      `

    default:
      return css``
  }
}
