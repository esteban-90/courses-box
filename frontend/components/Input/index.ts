export * from './Input'
export * from './Feedback'
