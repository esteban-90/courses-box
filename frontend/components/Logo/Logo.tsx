import styled from '@emotion/styled'

type LogoProps = {
  /** Logo size */
  size?: number
}

export const Logo = styled.header<LogoProps>`
  font-family: 'Monoton', cursive;
  font-size: ${({ size = 3 }) => `${size}rem`};
  color: ${({ theme }) => theme.font.logo};

  text-shadow: ${({ theme, size = 3 }) => {
    const { logoShadow1, logoShadow2 } = theme.font
    const $shadow = `
        0 0 ${0.1 * size}rem ${logoShadow1},
        0 0 ${0.05 * size}rem ${logoShadow2},
        0 0 ${0.07 * size}rem ${logoShadow2},
        0 0 ${0.08 * size}rem ${logoShadow2},
        0 0 ${0.1 * size}rem ${logoShadow2};
      `

    return $shadow
  }};
`
