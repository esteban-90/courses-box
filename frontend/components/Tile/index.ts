export * from './Tile'
export * from './CenteredTile'
