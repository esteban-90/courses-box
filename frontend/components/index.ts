export * from './Button'
export * from './Checkbox'
export * from './Course'
export * from './Icon'
export * from './IconButton'
export * from './Input'
export * from './Layout'
export * from './Link'
export * from './Logo'
export * from './Switch'
export * from './Tile'
