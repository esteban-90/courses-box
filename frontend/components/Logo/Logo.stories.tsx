import { ComponentStoryObj, ComponentMeta } from '@storybook/react'
import { expect } from '@storybook/jest'
import { screen } from '@storybook/testing-library'
import { Logo } from '@/components/Logo'

export default {
  title: 'Content/Logo',
  component: Logo,
} as ComponentMeta<typeof Logo>

export const BasicLogo: ComponentStoryObj<typeof Logo> = {
  async play() {
    await expect(screen.getByRole('banner')).toBeInTheDocument()
  },

  args: {
    children: 'CoursesBox',
  },
}
