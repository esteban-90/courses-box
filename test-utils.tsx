import { FC, ReactElement, ReactNode } from 'react'
import { render, RenderOptions, RenderResult } from '@testing-library/react'
import { ThemeProvider } from '@emotion/react'
import { Themes } from '@/styles/themes'

const Wrapper: FC<{ children: ReactNode }> = ({ children }): JSX.Element => {
  return <ThemeProvider theme={Themes.light}>{children}</ThemeProvider>
}

const customRender = (ui: ReactElement, options?: RenderOptions): RenderResult => {
  return render(ui, { wrapper: Wrapper, ...options })
}

export * from '@testing-library/react'
export { customRender as render }

// 14
