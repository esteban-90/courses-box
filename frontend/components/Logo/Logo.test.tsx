import { render } from '@/test-utils'
import { Logo } from '@/components/Logo'

describe('Logo test cases', () => {
  it('Render check', () => {
    const element = <Logo>CoursesBox</Logo>
    const { asFragment } = render(element)

    expect(asFragment()).toMatchSnapshot()
  })

  it('Render with custom size', async () => {
    const element = <Logo size={10}>CoursesBox</Logo>
    const { asFragment } = render(element)

    expect(asFragment()).toMatchSnapshot()
  })
})
