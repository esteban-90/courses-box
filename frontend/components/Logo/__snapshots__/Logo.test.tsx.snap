// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo test cases Render check 1`] = `
<DocumentFragment>
  <header
    class="css-qejboe"
  >
    CoursesBox
  </header>
</DocumentFragment>
`;

exports[`Logo test cases Render with custom size 1`] = `
<DocumentFragment>
  <header
    class="css-pjv2p7"
    size="10"
  >
    CoursesBox
  </header>
</DocumentFragment>
`;
